pybamm/models/submodels/interface/open_circuit_potential/wycisk_ocp.py

```diff
@@ -110,21 +96,28 @@ def get_coupled_variables(self, variables):
             variables[f"{Domain} electrode {phase_name}OCP hysteresis [V]"] = H
 
             # determine dQ/dU
-            def Q(sto, epsilon_s_av):
-                """Capacity change as a function of stoichiometry and active material volume fraction"""
-                c_max = self.phase_param.c_max
-                # epsilon_s_av = self.epsilon_s_av
-                epsilon_s_av = pybamm.yz_average(epsilon_s_av)
-                V_electrode = self.phase_param.main_param.A_cc * self.domain_param.L
-                Li_max = c_max * V_electrode * epsilon_s_av
-                Q_max = Li_max * self.phase_param.main_param.F / 3600
-                return Q_max * sto
+            # def Q(sto, epsilon_s_av):
+            #     """Capacity change as a function of stoichiometry and active material volume fraction"""
+            #     c_max = self.phase_param.c_max
+            #     # epsilon_s_av = self.epsilon_s_av
+            #     epsilon_s_av = pybamm.yz_average(epsilon_s_av)
+            #     V_electrode = self.phase_param.main_param.A_cc * self.domain_param.L
+            #     Li_max = c_max * V_electrode * epsilon_s_av
+            #     Q_max = Li_max * self.phase_param.main_param.F / 3600
+            #     return Q_max * sto
+
+            def Q(sto):
+                """Capacity change as a function of stoichiometry"""
+                if phase_name == "":
+                    C = variables[f"{Domain} electrode capacity [A.h]"]
+                else:
+                    C = variables[
+                        f"{Domain} electrode {phase_name}phase capacity [A.h]"
+                    ]
+                return C * sto
 
             dU = self.phase_param.U(sto_surf, T_bulk).diff(sto_surf)
-            epsilon_s_av = variables[
-                f"X-averaged {domain} electrode {phase_name}active material volume fraction"
-            ]
-            dQ = Q(sto_surf, epsilon_s_av).diff(sto_surf)
+            dQ = Q(sto_surf).diff(sto_surf)
```

Comment:
by the way, this will just return `C` (`(a * b).diff(a) = b`)

Reply:
I can give it a try without the function and differentiation. I remember it not working but I could be wrong.

Reply:
looks like there are no errors